### ISSUE TITLE
garbled text in ckpt file during training

### ISSUE THREAD

**Issue by Jenifer-Qin** [Noble-Lab/casanovo#490]
## Describe the Issue
I'm trying to use the training function to generate new model weight. It failed multiple times when I used my own data, so then I want to use your training data to try again. However, when I used your data from MSV000090982, it still failed. For each epoch, it shows NaN and says there is something wrong with the batch size. After training, I only get 3 ckpt files out of 30 epochs. And each of these 3 files are composed of garbled text that I can't read at all. Where should I change batch size? Or is there anything else that I made mistakes for training new model?

## Steps To Reproduce
prompt:  (casanovo_env02) C:\Users\SVC-LiLab-SOP>casanovo train -p E:/MSV90982-update/BC-1.mgf -c E:/casanovo_config3.yaml E:/MSV90982-update/BC-2.mgf
For this yaml file, I only edited the model_save_folder_path and val_check_interval to be 4_000.

## Expected Behavior
I don't expect to see NaNs during training, and I expect to get normal and proper model weight files.

## Terminal Output (If Applicable)
Part of the model weight generated from training and part of training concole is shown below

```
PK                      archive/data.pklFB ZZZZZZZZZZZZZZ€}q (X   epochqKX
   global_stepqMà.X   pytorch-lightning_versionqX
   2.5.0.post0qX
   state_dictqccollections
OrderedDict
q)Rq(X   encoder.latent_spectrumqctorch._utils
_rebuild_tensor_v2
q	((X   storageq
ctorch
FloatStorage
q
X   0qX   cpuq
M tqQK KKM ‡qM M K‡q‰h)RqtqRqX(   encoder.peak_encoder.mz_encoder.sin_termqh	((h
h
X   1qh
M tqQK M …qK…q‰h)RqtqRqX(   encoder.peak_encoder.mz_encoder.cos_termqh	((h
h
X   2qh
M tqQK M …qK…q ‰h)Rq!tq"Rq#X'   encoder.peak_encoder.int_encoder.weightq$h	((h
h
X   3q%h
M tq&QK M K†q'KK†q(‰h)Rq)tq*Rq+X=   encoder.transformer_encoder.layers.0.self_attn.in_proj_weightq,h	((h
h
X   4q-h
J   tq.QK M M †q/M K†q0‰h)Rq1tq2Rq3X;   encoder.transformer_encoder.layers.0.self_attn.in_proj_biasq4h	((h
h
X   5q5h
M tq6QK M …q7K…q8‰h)Rq9tq:Rq;X>   encoder.transformer_encoder.layers.0.self_attn.out_proj.weightq<h	((h
h
X   6q=h
J   tq>QK M M †q?M K†q@‰h)RqAtqBRqCX<   encoder.transformer_encoder.layers.0.self_attn.out_proj.biasqDh	((h


(casanovo_env02) C:\Users\SVC-LiLab-SOP>casanovo train -p E:/MSV90982-update/BC-1.mgf -c E:/casanovo_config3.yaml E:/MSV90982-update/BC-2.mgf
WARNING: Dataloader multiprocessing is currently not supported on Windows or MacOS; using only a single thread.
Seed set to 454
INFO: Casanovo version 4.3.0
INFO: Training a model from:
INFO:   E:/MSV90982-update/BC-2.mgf
INFO: Using the following validation files:
INFO:   E:/MSV90982-update/BC-1.mgf
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
INFO: Reading 1 files...
E:\MSV90982-update\BC-2.mgf: 16079spectra [00:14, 1118.72spectra/s]
INFO: Reading 1 files...
E:\MSV90982-update\BC-1.mgf: 15941spectra [00:16, 961.49spectra/s]

  | Name       | Type             | Params | Mode
--------------------------------------------------------
0 | encoder    | SpectrumEncoder  | 18.9 M | train
1 | decoder    | PeptideDecoder   | 28.4 M | train
2 | softmax    | Softmax          | 0      | train
3 | celoss     | CrossEntropyLoss | 0      | train
4 | val_celoss | CrossEntropyLoss | 0      | train
--------------------------------------------------------
47.4 M    Trainable params
0         Non-trainable params
47.4 M    Total params
189.407   Total estimated model params size (MB)
233       Modules in train mode
0         Modules in eval mode
Epoch 0:   0%|                                                                                 | 0/503 [00:00<?, ?it/s]WARNING: C:\Users\SVC-LiLab-SOP\AppData\Local\anaconda3\envs\casanovo_env02\lib\site-packages\lightning\pytorch\utilities\data.py:79: Trying to infer the `batch_size` from an ambiguous collection. The batch size we found is 32. To avoid any miscalculations, use `self.log(..., batch_size=batch_size)`.

WARNING: C:\Users\SVC-LiLab-SOP\AppData\Local\anaconda3\envs\casanovo_env02\lib\site-packages\lightning\pytorch\utilities\data.py:79: Trying to infer the `batch_size` from an ambiguous collection. The batch size we found is 32. To avoid any miscalculations, use `self.log(..., batch_size=batch_size)`.

Epoch 0: 100%|████████████████████████████████████████████████████████████▉| 502/503 [31:56<00:03,  0.26it/s, v_num=32]WARNING: C:\Users\SVC-LiLab-SOP\AppData\Local\anaconda3\envs\casanovo_env02\lib\site-packages\lightning\pytorch\utilities\data.py:79: Trying to infer the `batch_size` from an ambiguous collection. The batch size we found is 15. To avoid any miscalculations, use `self.log(..., batch_size=batch_size)`.

WARNING: C:\Users\SVC-LiLab-SOP\AppData\Local\anaconda3\envs\casanovo_env02\lib\site-packages\lightning\pytorch\utilities\data.py:79: Trying to infer the `batch_size` from an ambiguous collection. The batch size we found is 15. To avoid any miscalculations, use `self.log(..., batch_size=batch_size)`.

Epoch 0: 100%|█████████████████████████████████████████████████████████████| 503/503 [31:57<00:00,  0.26it/s, v_num=32]INFO: Step       Train loss      Valid loss
INFO: 503       2.961934        nan
```

**Comment by bittremieux** [comment]
The weights files are binary files, and are not intended to be opened manually. So the "garbled text" is as expected.

During training, training and validation losses are interleaved in the output. So it first reports a training loss with `nan` validation loss, and then a validation loss with `nan` training loss. This is a bit confusing, but because the losses are produced by lower-level dependencies, we can't easily change this behavior.

There is no issue with the batch size, just a warning that this was automatically inferred from the data. This is fine, and will be addressed in the next Casanovo release.

So I don't see any issue here, and you should be able to use the weights file you obtained to make predictions.
